hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java

```diff
@@ -69,8 +69,21 @@ protected Set<Key> allKeysForTransaction(
         // note, payerAccount should never be null, but we're dealing with Sonar here.
         final AccountID payerForNested =
                 scheduleInState.payerAccountIdOrElse(scheduleInState.schedulerAccountIdOrThrow());
-        final TransactionKeys keyStructure = context.allKeysForTransaction(scheduledAsOrdinary, payerForNested);
-        return getKeySetFromTransactionKeys(keyStructure);
+        try {
+            // check keys for nested transaction
+            final TransactionKeys keyStructure = context.allKeysForTransaction(scheduledAsOrdinary, payerForNested);
+            return getKeySetFromTransactionKeys(keyStructure);
+        } catch (PreCheckException e) {
+            // if the nested transaction is invalid, we return UNRESOLVABLE_REQUIRED_SIGNERS
+            // instead of whatever the nested transaction returned.
+            if (e.responseCode() == ResponseCodeEnum.INVALID_ACCOUNT_ID
+                    || e.responseCode() == ResponseCodeEnum.INVALID_TOKEN_ID
+                    || e.responseCode() == ResponseCodeEnum.INVALID_TOPIC_ID) {
```

Comment:
None of these should be failing.  Any exception thrown during key gathering generally is an error in the service.  The reason is that we're just asking for keys, not validating that the transaction can execute (this is why PureChecks cannot be called as part of the dispatch of allKeysForTransaction; I have removed that in my branch).
The Scheduled child transaction may not be valid when created, but valid when executed (an indeterminate amount of time later).
Unresolvable required signers is supposed to only be connected to missing accounts (although I think monoservice attaches it to some other fairly random causes).

Here is what I have on these lines:
```java
        // @todo('9447') We have an issue here.  Currently, allKeysForTransaction calls pureChecks and
        //                throws.  It *must not* throw and *should not* call pureChecks.  We need the keys, even
        //                if the transaction is invalid, because mono service creates and signs schedules for
        //                invalid transactions, then only fails when the transaction is executed.  This is
        //                probably incorrect, but we still have to match mono service for now.
        //                Over a dozen HapiTests, and some user applications, depend on this call not failing.
        try {
            final TransactionKeys keyStructure = context.allKeysForTransaction(scheduledAsOrdinary, payerForNested);
            return getKeySetFromTransactionKeys(keyStructure);
        } catch (final PreCheckException translated) {
            // Mono ignores nearly all errors in gathering required keys, but we cannot entirely do that.
            // For now, this breaks a lot of tests, because too many preHandle methods check a bit too carefully
            // so we don't get the keys gathered.  That's a change to all the services, however, so it has to
            // just fail for now, until more services are finished matching Monoservice behavior.
            throw new PreCheckException(translated.responseCode());
        }
```

Here is my change to PreHandleContextImpl (~line 451) (I have a similar change in HandleContextImpl):
```java
        // @todo('9447') Cannot call pure checks currently, as it does not match mono behavior.
        //           dispatcher.dispatchPureChecks(nestedTxn);
```

Reply:
Note, this is out of date and pureChecks was added back in.  failures due to pureChecks are fine, as the transaction will never be valid; it is just the failures during preHandle that cause issues.